service/src/main/java/datawave/microservice/authorization/OAuthControllerV2.java

```diff
@@ -0,0 +1,85 @@
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * Presents the REST operations for the authorization service to implement the OAuth2 code flow.
+ */
+@Tag(name = "OAuth Operations /v2",
+                externalDocs = @ExternalDocumentation(description = "Authorization Service Documentation",
+                                url = "https://github.com/NationalSecurityAgency/datawave-authorization-service"))
+@RestController
+@RequestMapping(path = "/v2/oauth", produces = MediaType.APPLICATION_JSON_VALUE)
+public class OAuthControllerV2 {
+    
+    private final OAuthOperationsV2 oauthOperations;
+    
+    @Autowired
+    public OAuthControllerV2(OAuthOperationsV2 oauthOperations) {
+        this.oauthOperations = oauthOperations;
+    }
+    
+    @Operation(summary = "Authorizes the calling user to produce a JWT value",
+                    description = "The returned JWT can be passed to other calls in a header. For example: \"Authorization: bearer <JWT value>\".<br>"
+                                    + "The user can be determined with from the supplied client certificate or trusted headers ("
+                                    + "X-SSL-clientcert-subject/X-SSL-clientcert-issuer).")
+    @RequestMapping(path = "/authorize", method = RequestMethod.GET)
+    public void authorize(@AuthenticationPrincipal DatawaveUserDetails currentUser, HttpServletResponse response, @RequestParam String client_id,
+                    @RequestParam String redirect_uri, @RequestParam String response_type, @RequestParam(required = false) String state)
+                    throws IllegalArgumentException, IOException {
+        oauthOperations.authorize(currentUser, response, client_id, redirect_uri, response_type, state);
+    }
+    
+    @Operation(summary = "Authorizes the calling user to produce a JWT value",
+                    description = "The returned JWT can be passed to other calls in a header. For example: \"Authorization: bearer <JWT value>\".<br>"
+                                    + "The user can be determined with from the supplied client certificate or trusted headers ("
+                                    + "X-SSL-clientcert-subject/X-SSL-clientcert-issuer).")
+    @RequestMapping(path = "/token", method = RequestMethod.POST)
+    public OAuthTokenResponse token(@AuthenticationPrincipal DatawaveUserDetails currentUser, HttpServletResponse response, @RequestParam String grant_type,
+                    @RequestParam String client_id, @RequestParam String client_secret, @RequestParam(required = false) String code,
+                    @RequestParam(required = false) String refresh_token, @RequestParam(required = false) String redirect_uri) throws IOException {
+        return oauthOperations.token(currentUser, response, grant_type, client_id, client_secret, code, refresh_token, redirect_uri);
+    }
+    
+    /**
+     * Returns the {@link DatawaveUserDetails} that represents the authenticated calling user.
```

Comment:
The comments for this method and the following method seem inaccurate: return type, descriptions are the same even though only I think only "users" returns a collection that includes proxies.

Reply:
I can't really speak to the validity of the documentation.  I simply lifted this from OAuthOperationsV2 and moved it to OAuthControllerV2.  Bill might have an opinion about the comments.  This is what we currently have on main FWIW.

Reply:
ok